presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -841,31 +843,36 @@ private List<Partition> batchGetPartition(String databaseName, String tableName,
     @Override
     public MetastoreOperationResult addPartitions(MetastoreContext metastoreContext, String databaseName, String tableName, List<PartitionWithStatistics> partitions)
     {
+        List<CompletableFuture<Future<BatchCreatePartitionResult>>> futures = new ArrayList<>();
```

Comment:
What is the performance benefit of changing this function? Can you run benchmarking with and without this optimization?

Earlier, the code as doing these in parallel as well. Once you create a list of futures, they all start running in parallel. So there should not be any difference if you await on them in serial or parallel as we want all to finish.

Reply:
I add more comments near code blocks to explain, but the idea is to create partition request via BatchCreatePartitionResult as a CompletableFuture, so it won't be a blocking call. The previous version of logic does batch-by-batch fashion instead of all-by-once fashion.

Reply:
Previous version isn't batch-by-batch. Futures are created by executorService.submit(), so computation has started in background as soon as you call batchCreatePartitionAsync().

Reply:
yes the computation is kicked off via async request but is it true that previous version at the end wait and check for the callback one-by-one? It makes more sense to optimize this code using `CompletableFuture` interface and wait for callbacks to complete at the end via `CompletableFuture.allOf`.

Reply:
@fgwang7w yes it checks results one by one, but it wouldn't make a difference as calling get() one by one is same as using a list of completable futures. This smells like micro-optimization. Can you run the benchmark without this?

Reply:
Just tested sync partition code path against 39k partitions with and without this code block change, and the result is very close. Per offline discussion, I will rollback this code block change to make the code more readable instead of using CompletableFuture